### PR TITLE
[stable/rabbitmq] Change default clustering method to hostname

### LINKED ISSUE #7806: [stable/rabbitmq] Persistent storage inconsistency
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

**Version of Helm and Kubernetes**:
1.10

**Which chart**:
stable/rabbitmq

**What happened**:
mnesia storage directory changes every time pod gets new IP address. Same happens for logs. If you will check persistent storage for any cluster that runs long enough (months) you will notice a lot of log files and mnesia storage directories named with ex-IPs of the pod.
Also there are high chances to actually lose data in case of dare situation where all pods of rabbitmq will get terminated at once.

**What you expected to happen**:
Since we're using statefulset for rabbitmq persistent storage should be in line with that. So for naming we should use `POD_NAME` rather than `POD_IP`.

**How to reproduce it** (as minimally and precisely as possible):
1. In case you don't have any additional vHost other than default one defined in helm configuration try to create additional vHost
2. delete all rabbitmq pods at once
3. Additional vHost from step #1 will disappear when cluster is back online.

**Anything else we need to know**:
Chart uses IP based clustering and with current configuration of rabbitmq it potentially may lead to rabbitmq inconsistency or even data/schema loss in case all the pods will go down all together.
Also during a lifetime a lot of directories of mnesia storage created with IPs those were allocated to the pod in name. Also it's quite inconvenient to locate RMQ node by IP address

### LINKED ISSUE #10841: [stable/rabbitmq] Persistence not working on GKE and AKS
**Is this a request for help?**: Yes

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Bug report

**Version of Helm and Kubernetes**: Helm 2.11.0, K8S 1.10.11-gke.1

**Which chart**: stable/rabbitmq

**What happened**:

I installed RabbitMQ with persistence enabled. After a node failure due to some far too heavy load being deployed, RabbitMQ automatically came back up, but all users were gone, all exchanges were gone and all queues were gone. I had to manually recreate all the users and exchanges.

**What you expected to happen**:

Recovery with all persistent data intact.

**How to reproduce it** (as minimally and precisely as possible):

1) Deploy stable/rabbitmq on a small cluster
2) Crash cluster by deploying something stupid like a massive neural network without limitations on resources
3) Delete offending pod, wait for recovery - persistent data is gone.

**Anything else we need to know**:

Values.yaml used;

```
rabbitmq:
  username: XXXXXX
  plugins: |-
      [rabbitmq_management,rabbitmq_peer_discovery_k8s].

  clustering:
    address_type: ip
    k8s_domain: cluster.local

  configuration: |-
      ## Clustering
      cluster_formation.peer_discovery_backend  = rabbit_peer_discovery_k8s
      cluster_formation.k8s.host = kubernetes.default.svc.cluster.local
      cluster_formation.node_cleanup.interval = 10
      cluster_formation.node_cleanup.only_log_warning = false
      cluster_partition_handling = autoheal
      ## queue master locator
      queue_master_locator=min-masters
      ## enable guest user
      loopback_users.guest = false

persistence:
  enabled: true
  accessMode: ReadWriteOnce
  size: 8Gi

resources:
  requests:
    memory: 256Mi
    cpu: 100m

replicas: 2

nodeSelector:
  beta.kubernetes.io/arch: amd64

ingress:
  enabled: false

metrics:
  enabled: true
```

### PR DESCRIPTION
**Pull request by tompizmor**
Signed-off-by: tompizmor <tompizmor@gmail.com>

#### What this PR does / why we need it:

Changes the default clustering address type from `ip` to `hostname`. 

#### Which issue this PR fixes

  - Fixes https://github.com/helm/charts/issues/10841
  - Fixes https://github.com/helm/charts/issues/7806

#### Special notes for your reviewer:
Don't merge it at the moment.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md

### CODE REVIEW

File: stable/rabbitmq/values.yaml [hunk 1]
```diff
@@ -139,7 +139,7 @@ securityContext:
 
 persistence:
   ## this enables PVC templates that will create one per pod
-  enabled: false
+  enabled: true
```
Comment (thomas-riccardi) [inline stable/rabbitmq/values.yaml:147]: do we want to enable persistence by default now? is it standard in other `stable/` charts?
if so, we should at least update the default value in `README.md` too.

Comment (tompizmor) [reply stable/rabbitmq/values.yaml:147]: I don't know why it was disabled before but yes, I think we should enable it. 
I will update the value in the README

Comment (thomas-riccardi) [reply stable/rabbitmq/values.yaml:147]: OK.
I did a quick `grep` and a large majority of charts enable persistence by default.

**Review by juan131 — APPROVED** [review]
/ok-to-test
/lgtm

**Pull request merged**